DeepSpeech.py

```diff
@@ -593,9 +595,23 @@ def __call__(self, progress, data, **kwargs):
             # Batch loop
             while True:
                 try:
-                    _, current_step, batch_loss, problem_files, step_summary = \
-                        session.run([train_op, global_step, loss, non_finite_files, step_summaries_op],
-                                    feed_dict=feed_dict)
+                    try:
+                        _, current_step, batch_loss, problem_files, step_summary = \
+                            session.run([train_op, global_step, loss, non_finite_files, step_summaries_op],
+                                        feed_dict=feed_dict)
+                    except errors_impl.InvalidArgumentError as err:
```

Comment:
Don't use the private import, this should be `tf.errors.InvalidArgumentError`.

Reply:
Also, I think this should just log the error, ignore that step and continue. I'm pretty sure when you retry below it's not actually retrying anything, it's moving on to the next batch, as the iterator has already consumed the current batch and can't be rewinded.